lib/KModal.vue

```diff
@@ -1,102 +1,104 @@
 <template>
 
-  <component :is="wrapper">
-    <!-- Accessibility properties for the overlay -->
-    <transition name="modal-fade" appear>
-      <div
-        id="modal-window"
-        ref="modal-overlay"
-        class="modal-overlay"
-        @keyup.esc.stop="emitCancelEvent"
-        @keyup.enter="handleEnter"
-      >
-        <!-- KeenUiSelect targets modal by using div.modal selector -->
+  <KFocusTrap>
+    <component :is="wrapper">
```

Comment:
We'll need to test that when `component` here is `KOverlay` (that is, when `KModal`'s `appendToOverlay` prop is `true`), the focus trapping still works as expected.

Just from the code I'm a bit suspicious there may be trouble. Reason being that `KOverlay` will move the modal content to another element `#k-overlay`, whereas `KFocusTrap` will stay in its original place in the DOM.

I know I used this structure as an example in the issue, however it was meant to be just rough guidance rather than solution, apologies if that was confusing. Let's preview for both possible `appendToOverlay` values and see if it needs to be adjusted.

Reply:
oh , i should have understood the requirements properly, thanks @MisRob  i will make sure to use KFocusTrap appropriately.